### ISSUE TITLE
External IPs are not displayed after SUC upgrade

### ISSUE THREAD

**Issue by mdrahman-suse** [rancher/rke2#3000]
**Environmental Info:**
RKE2 Version: 

rke2 version v1.24.1-rc1+rke2r1 (aa3027502617aa492628da505111107acecde0ce)
go version go1.18.1b7

Node(s) CPU architecture, OS, and Version: 

Ubuntu SMP Sun Apr 12 19:46:01 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:

3 servers, 2 agents

**Describe the bug:**

External IPs are not getting displayed after upgrading the cluster via SUC upgrade.

**Steps To Reproduce:**

- Installed RKE2 with version v1.23.6+rke2r2 with a 3 server, 2 worker configuration
- Ensure nodes, pods are up and running and external IPs are displayed on `kubectl get nodes,pods -A -o wide`
- Deploy nodeport.yaml, ingress.yaml, suc.yaml
- Ensure external IPs are displayed on `kubectl get all,ingress -o wide`
- Perform SUC upgrade
- Wait for the nodes to be upgraded
- Once upgrade is complete, run the commands again

NOTE: This may be a minor issue, but raising it for visibility

**Expected behavior:**

After the upgrade, the external IPs are visible

**Actual behavior:**

After the upgrade, the external IPs are not visible

**Additional context / logs:**

The cluster continues to function as expected so no specific errors were captured.

- Before the upgrade

```
NAME            STATUS   ROLES                       AGE   VERSION          INTERNAL-IP     EXTERNAL-IP      OS-IMAGE           KERNEL-VERSION   CONTAINER-RUNTIME
node/agent2     Ready    <none>                      13m   v1.23.6+rke2r2   xxx.yy.zz.233   x.yy.zz.191      Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.5.11-k3s2
node/agent1     Ready    <none>                      13m   v1.23.6+rke2r2   xxx.yy.zz.75    x.yy.zzz.29      Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.5.11-k3s2
node/server3    Ready    control-plane,etcd,master   17m   v1.23.6+rke2r2   xxx.yy.z.196    xx.yyy.zzz.112   Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.5.11-k3s2
node/server2    Ready    control-plane,etcd,master   15m   v1.23.6+rke2r2   xxx.yy.z.148    x.yyy.zzz.76     Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.5.11-k3s2
node/server1    Ready    control-plane,etcd,master   14m   v1.23.6+rke2r2   xxx.yy.z.52     xx.yyy.zz.162    Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.5.11-k3s2
```

- After the upgrade

```
NAME            STATUS   ROLES                       AGE   VERSION          INTERNAL-IP     EXTERNAL-IP   OS-IMAGE           KERNEL-VERSION   CONTAINER-RUNTIME
node/agent2     Ready    <none>                      31m   v1.24.1+rke2r1   xxx.yy.zz.233   <none>        Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/agent1     Ready    <none>                      31m   v1.24.1+rke2r1   xxx.yy.zz.75    <none>        Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/server3    Ready    control-plane,etcd,master   35m   v1.24.1+rke2r1   xxx.yy.z.196    <none>        Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/server2    Ready    control-plane,etcd,master   33m   v1.24.1+rke2r1   xxx.yy.z.148    <none>        Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/server1    Ready    control-plane,etcd,master   32m   v1.24.1+rke2r1   xxx.yy.z.52     <none>        Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
```

CC: @ShylajaDevadiga

**Comment by manuelbuil** [comment]
When using version `v1.24.1+rke2r1` I confirm that the external-ip config is ignored

**Comment by brandond** [comment]
Appears to be caused by the same changes as https://github.com/k3s-io/k3s/pull/5579 - 1.24 changes the behavior around how the kubelet and cloud controller work together to set the node IPs:
* https://github.com/kubernetes/kubernetes/pull/107750

**Comment by mdrahman-suse** [comment]
Verified the fix on RC v1.24.1-rc2+rke2r1
```
$ rke2 -v
rke2 version v1.24.1-rc2+rke2r1 (287923bd9e58c38180f70a914c864d382c4cbc35)
go version go1.18.1b7
```
- Fresh install
```
$ kubectl get nodes, pods -A -o wide
NAME                   STATUS   ROLES                       AGE   VERSION          INTERNAL-IP    EXTERNAL-IP     OS-IMAGE           KERNEL-VERSION   CONTAINER-RUNTIME
node/ip-172-31-12-87   Ready    control-plane,etcd,master   13m   v1.24.1+rke2r1   xxx.yy.zz.87   x.yyy.zzz.127   Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/ip-172-31-14-44   Ready    <none>                      11m   v1.24.1+rke2r1   xxx.yy.zz.44   x.yyy.zzz.67    Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/ip-172-31-4-182   Ready    control-plane,etcd,master   15m   v1.24.1+rke2r1   xxx.yy.z.182   x.yyy.zzz.58    Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/ip-172-31-6-17    Ready    control-plane,etcd,master   13m   v1.24.1+rke2r1   xxx.yy.z.17    x.yyy.zz.108    Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
```
- Upgrade
```
$ kubectl get nodes,pods -A -o wide
NAME                   STATUS   ROLES                       AGE   VERSION          INTERNAL-IP    EXTERNAL-IP      OS-IMAGE           KERNEL-VERSION   CONTAINER-RUNTIME
node/ip-172-31-0-106   Ready    <none>                      54m   v1.24.1+rke2r1   xxx.yy.z.106   x.yyy.zz.8       Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/ip-172-31-5-227   Ready    control-plane,etcd,master   56m   v1.24.1+rke2r1   xxx.yy.z.227   xx.yyy.zzz.71    Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/ip-172-31-8-154   Ready    control-plane,etcd,master   58m   v1.24.1+rke2r1   xxx.yy.z.154   x.yyy.zz.138     Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
node/ip-172-31-9-215   Ready    control-plane,etcd,master   55m   v1.24.1+rke2r1   xxx.yy.z.215   xx.yyy.zzz.245   Ubuntu 20.04 LTS   5.4.0-1009-aws   containerd://1.6.4-k3s1
```
